godel_surface_detection/src/detection/surface_detection.cpp

```diff
@@ -433,5 +410,29 @@ namespace godel_surface_detection
 
       return name;
     }
+
+    void SurfaceDetection::filterFullCloud()
+    {
+      pcl::PointCloud<pcl::PointXYZRGB>::Ptr intermediate_cloud_ptr(new pcl::PointCloud<pcl::PointXYZRGB>);
+
+      //remove the table using the passthrough filter
+      pcl::PassThrough<pcl::PointXYZRGB> pass;
+      pass.setInputCloud(full_cloud_ptr_);
+      const std::string FILTER_DIRECTION = "z";
+      pass.setFilterFieldName (FILTER_DIRECTION);
+      //keep poin clouds with these limits
+      const double MINIMUM_DISTANCE = 0.01; // 1 cm
+      const double MAXIMUM_DISTANCE = 1.0; // 1 m
+      pass.setFilterLimits (MINIMUM_DISTANCE, MAXIMUM_DISTANCE);
+      pass.filter (*intermediate_cloud_ptr);
+
+      //downsample the full cloud using the voxelgrid filter method
+      pcl::VoxelGrid<pcl::PointXYZRGB> vox;
+      vox.setInputCloud (intermediate_cloud_ptr);
+      vox.setLeafSize (INPUT_CLOUD_VOXEL_FILTER_SIZE,
+                       INPUT_CLOUD_VOXEL_FILTER_SIZE,
+                       INPUT_CLOUD_VOXEL_FILTER_SIZE);
+      vox.filter(*process_cloud_ptr_);
+    }
   } /* end namespace detection */
```

Comment:
I know that sometimes it isn't possible but it'd be nice to keep the changes in PRs separate from one another, I found that the changes here also made it into PR #212 .  I'll just paste my comment in PR #212 about these changes here:
>>>This approach assumes that the table is very close to the robot base z = 0 , is this meant to be a permanent solution or just a temporary workaround? It seems that the findLargestCloud() method would've accomplished the same table removal task without making any assumptions about the tables z placement.